waspc/src/Wasp/Common.hs

```diff
@@ -1,6 +1,12 @@
 module Wasp.Common
   ( WaspProjectDir,
+    dbMigrationsDirInWaspProjectDir,
   )
 where
 
+import StrongPath (Dir', Path', Rel, reldir)
+
 data WaspProjectDir -- Root dir of Wasp project, containing source files.
+
+dbMigrationsDirInWaspProjectDir :: Path' (Rel WaspProjectDir) Dir'
```

Comment:
NIce, this looks like a good place for this path!

Just to double check, since I see no other paths here besides WaspProjectDir -> where are the other paths? Specifically, I know of `ext` and `.env` -> so first is external files and second is dotenv. Could you just double check where they are being defined? Maybe search project for "ext/" or "ext" or ".env". It might be that they are defined somewhere locally and were never extracted into Common because we never needed them in multiple places in which case that is fine, but I am just double checking in case they are all defined maybe somewhere at the same place and we don't know about it.

Reply:
Oh, regarding `Dir'` -> so you have a choice here, you can keep it as `Dir'` or you can make it `(Dir DbMigrationsDir)` where you would additionally define `data DbMigrationsDir` also in this file, in order to **name** the directory.

What does it mean to name it, why would you do it? Well, at ensures at type level that you are dealing with exactly this directory. For example, if there is a function that copies contents of db migrations dir from wasp source into generated project, you could say that it takes two arguments: `Path' (Rel WaspProjectDir) (Dir Wasp.Common.DbMigrationsDir)` and the second one `Path' (Rel ProjectRootDir) (Dir Wasp.Generator.Common.DbMigrationsDir)` and this way you ensure they get exactly these, correct paths, and no other path. Btw I might have missed some modules or smth here but I think you will get the idea. So in this case I would actually do that, name it, since we know there is exactly one such directory. While for example if you were reading a bunch of files/dir from somewhere, then you would use `File'` or `Dir'` for them since you don't really distinguish them in any way.

Reply:
I will double check on where other paths are defined to make sure it wasn't somewhere else vs not yet extracted into Common! 👍🏻 

Good call on making it explicitly `(Dir DbMigrationsDir)` to help avoid any issues of mixing the final dir type, I will make that change.

Reply:
The change from `Dir'` to `Dir DbMigrationsDir` can be found here: https://github.com/wasp-lang/wasp/pull/383/commits/3aa6b67e3c9f075487746d7e153de7ede4f54846 👍🏻 

In regard to other places where we have external paths defined, in `cli` most are in cli/Wasp/Cli/Common.hs. On the `src` side, the convention appears to be more using the *Generator.hs files or their corresponding /Generator/*Generator/Common.hs files and keeping the definitions pretty localized to their usage. So for this PR, I do not think we missed putting the topmost declarations into a more appropriate place than  and it feels like it is following the conventions.

In the future, we can consider a couple things:
- Define more common paths in src/Wasp/Common.hs we can leverage in both `cli` and `src` (also helpful as Generators do more Wasp dir work)
- Cleaning up .env by defining it in src/Wasp/Common.hs and corresponding usage in src/Wasp/Lib.hs. All other `Path' Abs (Dir WaspProjectDir)` usage is in `cli` right now.

Reply:
Ok, that all sounds good, including tasks for the future!